mdoc/Mono.Documentation/MDocUpdater.cs

```diff
@@ -3626,124 +3623,129 @@ static bool UpdateAssemblyVersionForAssemblyInfo (XmlElement e, XmlElement root,
+        /// <param name="fx"></param>
+        /// <param name="typeEntry"></param>
+        /// <param name="assemblyName"></param>
+        /// <param name="otherLangs"></param>
+        public static void MakeAttributes(
+            XmlElement root,
+            IEnumerable<(CustomAttribute, string)> customAttributesWithPrefix,
+            FrameworkEntry fx,
+            FrameworkTypeEntry typeEntry,
+            string assemblyName = null)
+        {
+            const string Language = "Language";
+
+            bool isAssemblyAttribute = !string.IsNullOrWhiteSpace(assemblyName);
+
+            XmlElement e = (XmlElement)root.SelectSingleNode("Attributes");
+
             if (e == null)
-                e = root.OwnerDocument.CreateElement ("Attributes");
-            
-            var attributesCurrent = attributes
-                .Select (FilterSpecialChars)
-                .Distinct () // make sure there aren't any dupes
-                .ToDictionary (a => a, a => false); // key is the attribute, value is whether it was found
-
-            var attributesState = e.ChildNodes.SafeCast<XmlElement> ()
-                                .Select (elem => new
-                                {
-                                    ExistingValue = elem.FirstChild.InnerText,
-                                    XElement = elem,
-            }).ToArray();
-
-
-            // iterate this framework's attributes and compare against the state
-            foreach(var currentAttribute in attributesCurrent)
-            {
-                var alreadyExists = attributesState.FirstOrDefault (state => state.ExistingValue == currentAttribute.Key);
-                if (alreadyExists != null)
-                {
-                    // TODO: add to unconditionally
-                    // if FXA, add fx.name
-                    if (fx.FrameworksCount > 1)// && alreadyExists.XElement.HasAttribute(Consts.FrameworkAlternate))
-                    {
-                        var newfxlist = FXUtils.AddFXToList (alreadyExists.XElement.GetAttribute (Consts.FrameworkAlternate), fx.Name);
-                        alreadyExists.XElement.SetAttribute (Consts.FrameworkAlternate, newfxlist);
-                    }
-                    continue;
-                }
-                else // let's create it
-                {
-                    XmlElement ae = root.OwnerDocument.CreateElement ("Attribute");
-                    e.AppendChild (ae);
-                    WriteElementText (ae, "AttributeName", currentAttribute.Key);
-
-                    if (fx.FrameworksCount > 1)
-                    {
-                        ae.SetAttribute (Consts.FrameworkAlternate, fx.Name);
-                    }
+                e = root.OwnerDocument.CreateElement("Attributes");
+
+            // we'll use C# version as key identifier of each attribute
+            var existingAttrElements = e.ChildNodes.SafeCast<XmlElement>()
```

Comment:
Ok ... so it looks these changes were based on an older method for writing monikerized values (the attributes were an older implementation), which just tries to update the existing state, rather than what we've moved to that uses a common method to deal with the complexity of managing the FXA attributes. This common method basically clears elements the first time it comes across it, and rebuilds it every time mdoc is run, which I've found to be a cleaner and less bug-prone method.

You can see an example of this method's use here ... you basically just have to pass in some values and delegates (which lets you customize how it works for different cases) and then it takes care of everything else.
https://github.com/mono/api-doc-tools/blob/develop/mdoc/Mono.Documentation/MDocUpdater.cs#L4185-L4206